paimon-core/src/test/java/org/apache/paimon/metrics/commit/CommitMetricsTest.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.metrics.commit;
+
+import org.apache.paimon.manifest.FileKind;
+import org.apache.paimon.manifest.ManifestEntry;
+import org.apache.paimon.metrics.Gauge;
+import org.apache.paimon.metrics.Histogram;
+import org.apache.paimon.metrics.Metric;
+import org.apache.paimon.metrics.MetricGroup;
+import org.apache.paimon.metrics.Metrics;
+import org.apache.paimon.utils.ExceptionUtils;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.manifest.ManifestFileMetaTestBase.makeEntry;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.offset;
+
+/** Tests for {@link CommitMetrics}. */
+public class CommitMetricsTest {
+    private static final String TABLE_NAME = "myTable";
+
+    private CommitMetrics commitMetrics;
+
+    @BeforeEach
+    public void beforeEach() {
+        commitMetrics = getCommitMetrics();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        commitMetrics.close();
+    }
+
+    /** Tests the registration of the commit metrics. */
+    @Test
+    public void testGenericMetricsRegistration() {
+        MetricGroup genericMetricGroup = commitMetrics.getMetricGroup();
+        AssertionError error = null;
+        try {
+            assertThat(Metrics.getInstance().getMetricGroups().size()).isEqualTo(1);
+        } catch (AssertionError e) {
+            error =
+                    ExceptionUtils.firstOrSuppressed(
+                            e,
+                            new AssertionError(
+                                    "Please close the created TableCommit objects in case of metrics resource leak."));
+        }
```

Comment:
```suggestion
        assertThat(Metrics.getInstance().getMetricGroups().size()).isEqualTo(1).withFailMessage(xxx);
```

`Please close the created TableCommit objects` -> `Please close the created metric groups`. In the future we're going to add more metrics aside from commit metrics.

You might want to add a helper method to print out all metric group objects, so developers can see what kinds of metric groups are not closed.